packages/sdk/src/dataProtector/sharing/removeFromCollection.ts

```diff
@@ -0,0 +1,97 @@
+import { GraphQLClient } from 'graphql-request';
+import { DEFAULT_SHARING_CONTRACT_ADDRESS } from '../../config/config.js';
+import { ErrorWithData, WorkflowError } from '../../utils/errors.js';
+import {
+  addressOrEnsOrAnySchema,
+  throwIfMissing,
+} from '../../utils/validators.js';
+import {
+  IExecConsumer,
+  RemoveFromCollectionParams,
+  SuccessWithTransactionHash,
+  SubgraphConsumer,
+  Address,
+} from '../types/index.js';
+import { getSharingContract } from './smartContract/getSharingContract.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
+
+export const removeFromCollection = async ({
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
+  protectedDataAddress = throwIfMissing(),
+}: IExecConsumer &
+  SubgraphConsumer &
+  RemoveFromCollectionParams): Promise<SuccessWithTransactionHash> => {
+  const vProtectedDataAddress = addressOrEnsOrAnySchema()
+    .required()
+    .label('protectedDataAddress')
+    .validateSync(protectedDataAddress);
+
+  const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+
+  const protectedData = await checkAndGetProtectedData({
+    graphQLClient,
+    protectedDataAddress: vProtectedDataAddress,
+    userAddress,
+  });
+
+  const sharingContract = await getSharingContract();
+  try {
+    const tx = await sharingContract.removeProtectedDataFromCollection(
+      protectedData.collection.id,
+      vProtectedDataAddress
+    );
+    await tx.wait();
+    return {
+      success: true,
+      txHash: tx.hash,
+    };
+  } catch (e) {
+    throw new WorkflowError('Failed to Remove Protected Data From Renting', e);
+  }
+};
+
+async function checkAndGetProtectedData({
+  graphQLClient,
+  protectedDataAddress,
+  userAddress,
+}: {
+  graphQLClient: GraphQLClient;
+  protectedDataAddress: Address;
+  userAddress: Address;
+}) {
+  const { protectedData } = await getProtectedDataById({
+    graphQLClient,
+    protectedDataAddress,
+  });
+
+  if (!protectedData) {
+    throw new ErrorWithData(
+      'This protected data does not exist in the subgraph.',
+      { protectedDataAddress }
+    );
+  }
+
+  if (protectedData.owner.id !== DEFAULT_SHARING_CONTRACT_ADDRESS) {
+    throw new ErrorWithData(
+      'This protected data is not owned by the sharing contract. First call addToCollection()',
+      {
+        protectedDataAddress,
+        currentOwnerAddress: protectedData.owner.id,
+      }
+    );
+  }
+
+  if (protectedData.collection?.owner?.id !== userAddress) {
+    throw new ErrorWithData(
+      'This protected data is not part of a collection owned by the user.',
+      {
+        protectedDataAddress,
+        currentCollectionOwnerAddress: protectedData.collection?.owner?.id,
+      }
+    );
+  }
+  // TODO : check => onlyCollectionNotSubscribed
+  // TODO : check => onlyProtectedDataNotRented
```

Comment:
Add something like this?

```
  const hasActiveRentals = protectedData.rentals.length > 0;

  if (hasActiveRentals) {
    throw new ErrorWithData('This protected data has active rentals.', {
      protectedDataAddress,
      activeRentalsCount: protectedData.rentals.length,
    });
  }
```

Reply:
cc20988a9a068d141af0a52e5fb44400cf44e26e